### PR TITLE
Only hold the mxs lock during init/close

### PR DESCRIPTION
**Pull request by texasaggie97**
- [X] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [X] ~~I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.~~
- [X] ~~I've added tests applicable for this pull request~~

### What does this Pull Request accomplish?
* Use function scope fixture to only hold the mxs (sim_db) lock during init and close, but not throughout the test
   * This is unfortunately needed due to concurrency bugs with DAQmx framework-based devices.
* Release the lock during the actual test

### What testing has been done?
* System

### CODE REVIEW

File: src/nifgen/system_tests/test_system_nifgen.py [hunk 1]
```diff
@@ -23,6 +23,15 @@ def session():
         yield simulated_session
```
Comment (marcoskirsch) [inline src/nifgen/system_tests/test_system_nifgen.py]: Please improve documentation where the locks are created.
Internal NI bug should be referenced.

File: src/niswitch/system_tests/test_system_niswitch.py [hunk 2]
```diff
@@ -19,6 +19,14 @@ def session():
         yield simulated_session
```
Comment (marcoskirsch) [inline src/niswitch/system_tests/test_system_niswitch.py]: please improve comment for locks

**Comment by texasaggie97** [comment]
test this please

File: src/nifgen/system_tests/test_system_nifgen.py (hunk 1 above)
Comment (texasaggie97) [reply src/nifgen/system_tests/test_system_nifgen.py]: Comment updated

File: src/niswitch/system_tests/test_system_niswitch.py (hunk 2 above)
Comment (texasaggie97) [reply src/niswitch/system_tests/test_system_niswitch.py]: Comment updated

**Pull request merged**